### ISSUE TITLE
`=` characters are stripped in environment variables values passed with `--env` option

### ISSUE THREAD

**Issue by Tom-Bonnike** [cypress-io/cypress#620]
Hi,

This only happens if the `=` is the last character of the variable. Could be very annoying with base64 environment variables.

### Versions
- Operating System: MacOS 10.12.5
- Cypress CLI: 0.13.1
- Cypress App: 0.19.4

### Is this a Feature or a Bug?
Bug 🐜

### How to reproduce

For a variable named `FOO` with a value of `BAR=`:
`./node_modules/.bin/cypress run ${PATH} --env FOO=BAR=`
`./node_modules/.bin/cypress run ${PATH} --env FOO='BAR='`
`./node_modules/.bin/cypress run ${PATH} --env FOO="BAR="`,
`./node_modules/.bin/cypress run ${PATH} --env CY=PRESS,FOO=BAR=`

`Cypress.config('FOO')` would return `BAR` and not `BAR=`, with all 4 ways. Using such variable in cypress.json or cypress.env.json doesn’t strip the `=`.

### Expected behavior
The `=` isn’t stripped.

**Comment by bahmutov** [comment]
This seems like a very limited edge case: having values of variables that include `=` characters. Is there a Node lib that can successfully parse and split values by `=` if some of the strings include `=`?

**Comment by HugoGiraudel** [comment]
While this is an edge case, this is still quite a bug and it does not seem too hard to fix. I don’t know your code base, but something as simple as the following should do the trick (no matter the language):

```js
const value = 'CY=PRESS,FOO=BAR='
const pairs = value.split(/\s*,\s*/)

pairs.forEach(pair => {
  const index = pair.indexOf('=')
  const key = pair.slice(0, index)
  const value = pair.slice(index + 1)

  console.log({ key, value })
})
```

Output:

```
{ key: "CY", value: "PRESS" }
{ key: "FOO", value: "BAR=" }
```

Am I missing something maybe?

**Comment by brian-mann** [comment]
I am thinking here you could just use the module API and not use it from the command line.

The module API accepts `env` as an object literal where you could specify this correctly.

https://docs.cypress.io/guides/guides/command-line.html#Cypress-Module-API

**Comment by HugoGiraudel** [comment]
Unfortunately, that is not really an option for us. We do rely on `cypress.json` though but still, the bug remains. :)

**Comment by brian-mann** [comment]
Another option is to use a different character other than `=` and then in your Cypress supportFile just iterate through them all and replace the special character with `=`.

That should also work.

```js
const obj = Cypress.env()

// loop
// replace special char with `=`
```

**Comment by HugoGiraudel** [comment]
That seems like a super hacky solution for something that should inherently be fixed. We pass a base64 encoded string through `--env` which contains an equal. This value comes from a different service. I’m not sure replacing `=` with another character only to restore equals slightly further down the pipe is a good thing.

Don’t get me wring, I really appreciate you providing workarounds for us. As I said, right now we rely on `cypress.json` which is fine-ish. But is there any reason this problem would be hard to fix for you?

**Comment by brian-mann** [comment]
We use https://github.com/substack/minimist to parse the args. If it doesn't support this out of the box then we'd have to add our own sanitation layer.

**Comment by HugoGiraudel** [comment]
To parse the args, yes. But once you have the value of the `--env` arg, I believe you parse it yourself right? I don’t think minimist provides a way to decompose arguments value on arbitrary syntax. Or am I missing something?

**Comment by brian-mann** [comment]
We will look at it. Worst case you'll have to wait until we open source and then just submit a PR. We are open sourcing at our next major release `1.0.0` which is happening soooooooon.

**Comment by HugoGiraudel** [comment]
Happy to try contributing when the code is open source. :)

**Comment by brian-mann** [comment]
Fixed in [`1.0.2`](https://docs.cypress.io/guides/references/changelog.html#1-0-2).
